qiskit/transpiler/target.py

```diff
@@ -838,7 +838,7 @@ def instruction_properties(self, index):
 
     def _build_coupling_graph(self):
         self._coupling_graph = rx.PyDiGraph(multigraph=False)
-        self._coupling_graph.add_nodes_from([{} for _ in range(self.num_qubits)])
+        self._coupling_graph.add_nodes_from([{}]* self.num_qubits)
```

Comment:
This is likely not what's intended.  This uses the same `{}` object for every element of the list, which means modifications to one affect all of them.

Reply:
(i.e. `xs = [x] * 2` satisfies `xs[0] is xs[1]` for all `x`)

Reply:
Yeah you're right I was hasty here

Reply:
Oh of course. That's a well known source of bugs. Still I'll audit for the case that the elements are immutable.